xdsl/dialects/arm/ops.py

```diff
@@ -0,0 +1,39 @@
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar
+
+from xdsl.ir import Operation
+from xdsl.irdl import IRDLOperation, irdl_op_definition, result_def
+
+from .register import IntRegisterType
+
+R1InvT = TypeVar("R1InvT", bound=IntRegisterType)
+
+
+class ARMOp(Operation, ABC):
+    """
+    Base class for operations that can be a part of ARM assembly printing.
+    """
+
+    @abstractmethod
+    def assembly_line(self) -> str | None:
+        raise NotImplementedError()
+
+
+class GetAnyRegisterOperation(Generic[R1InvT], IRDLOperation, ARMOp, ABC):
+    """
+    This instruction allows us to create an SSAValue for a given register name.
+    """
+
+    result = result_def(R1InvT)
+
+    def __init__(self, register_type: R1InvT):
+        super().__init__(result_types=[register_type])
+
+    def assembly_line(self) -> str | None:
+        return None
```

Comment:
If we're starting with specific ops, and generalizing later, it would make sense to move this into the GetRegisterOp for now, and to generalize when we add float registers.

Reply:
If there are immediate plans to add more register types, then I'd be happy with leaving it as is

Reply:
I'm not sure how immediate they are, and we may well want to put them in a different place when that happens, so it's not a bad idea to make it specific for the time being